cpp/src_prims/sparse/coo.h

```diff
@@ -934,60 +947,78 @@ __global__ static void symmetric_sum(int *restrict edges,
  */
 template <typename math_t, int TPB_X = 32, int TPB_Y = 32>
 void from_knn_symmetrize_matrix(const long *restrict knn_indices,
-                                const math_t *restrict knn_dists, const int n,
-                                const int k, COO<math_t> *out,
+                                const math_t *restrict knn_dists,
+                                const int n,
+                                const int k,
+                                /* math_t *restrict out, */
+                                math_t *restrict VAL,
+                                int *restrict COL,
+                                int *restrict ROW,
```

Comment:
I would strongly encourage the use of a templates type to replace the ints for the index types. I’ve been adjusting the other sparse prims as I’ve been encountering them in pull requests. The use of hard-coded int causes unexpected overflows when we begin to scale these algorithms.

Reply:
Oh ok very good idea :) I'll see if I can change int to a template arg